Detectors/ITSMFT/MFT/workflow/src/TrackFitterSpec.cxx

```diff
@@ -0,0 +1,175 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file TrackFitterSpec.cxx
+/// \brief Implementation of a data processor to read, refit and send tracks with attached clusters
+///
+/// \author Philippe Pillot, Subatech; adapted by Rafael Pezzi, UFRGS
+
+#include "MFTWorkflow/TrackFitterSpec.h"
+#include "DataFormatsITSMFT/Cluster.h"
+#include "Field/MagneticField.h"
+#include "TGeoGlobalMagField.h"
+#include "DetectorsBase/Propagator.h"
+
+#include <stdexcept>
+#include <list>
+
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/Lifetime.h"
+#include "Framework/Output.h"
+#include "Framework/Task.h"
+
+#include "MFTTracking/TrackParamMFT.h"
+#include "MFTTracking/TrackCA.h"
+#include "MFTTracking/FitterTrackMFT.h"
+#include "MFTTracking/TrackFitter.h"
+#include "MFTTracking/TrackExtrap.h"
+
+using namespace std;
+using namespace o2::framework;
+
+namespace o2
+{
+namespace mft
+{
+
+void TrackFitterTask::init(InitContext& ic)
+{
+  /// Prepare the track extrapolation tools
+  LOG(INFO) << "initializing track fitter";
+  mTrackFitter = std::make_unique<o2::mft::TrackFitter>();
+
+  auto filename = ic.options().get<std::string>("grp-file");
+  const auto grp = o2::parameters::GRPObject::loadFrom(filename.c_str());
+  if (grp) {
+    mGRP.reset(grp);
+    o2::base::Propagator::initFieldFromGRP(grp);
+    auto field = static_cast<o2::field::MagneticField*>(TGeoGlobalMagField::Instance()->GetField());
+
+    double centerMFT[3] = {0, 0, -61.4}; // Field at center of MFT
+    mTrackFitter->setBz(field->getBz(centerMFT));
+
+  } else {
+    LOG(ERROR) << "Cannot retrieve GRP from the " << filename.c_str() << " file !";
+    mState = 0;
+  }
+  mState = 1;
+}
+
+//_________________________________________________________________________________________________
+void TrackFitterTask::run(ProcessingContext& pc)
+{
+
+  if (mState != 1)
+    return;
+
+  auto tracksLTF = pc.inputs().get<const std::vector<o2::mft::TrackLTF>>("tracksltf");
+  auto tracksCA = pc.inputs().get<const std::vector<o2::mft::TrackCA>>("tracksca");
```

Comment:
both TrackCA and TrackLTF are POD classes, instead of ``get<const std::vector<...>>`` one should use ``get<gsl::span<...>>`` (one can do this in separate PR).